### ISSUE TITLE
Error when .Rmd file is removed then added back in repository without committing

### ISSUE THREAD

**Issue by pcarbo** [workflowr/workflowr#45]
I get this error:

```
Error in git2r::blame(repo, f) :
Error in 'git2r_blame_file': the path 'index.Rmd' does not exist in the given tree
```

Perhaps this error could be handled a bit more gracefully.

I labeeled this Issue as a bug, but perhaps could be described as an enhancement.

**Comment by jdblischak** [comment]
The good news is that this should only happen if users remove the .Rmd at the command line. If they use the new workflowr function `wflow_remove()` (introduced in [v0.6.0](https://jdblischak.github.io/workflowr/news/index.html#workflowr-0-6-0)), both the Rmd and HTML files will be removed from the Git repo in the same commit.

I'll look into how difficult it will be to catch this error before the file gets passed to git2r.

**Comment by jdblischak** [comment]
I think this situation is handled better now. In the situation where the previously committed file was removed from the Git repository and then recreated, but its HTML remains committed, workflowr now has the following behavior:

* `wflow_build()` runs without error
* `wflow_status()` provides a warning message and solution
* `wflow_publish()` solves the issue

Here's reprex of the current behavior. The warning message from `wflow_status()` was added in the previous release (1.6.0).

``` r
# Setup
cwd <- getwd()
d <- tempfile()
wflow_start(d)
#> wflow_start:
#> - New directory created at /tmp/RtmpegncBz/file7b604db76588
#> - Project name is "file7b604db76588"
#> - Working directory changed to /tmp/RtmpegncBz/file7b604db76588
#> - Git repo initiated at /tmp/RtmpegncBz/file7b604db76588
#> - Files were committed in version a0f57b0
wflow_publish("analysis/*Rmd")
#> Current working directory: /tmp/RtmpegncBz/file7b604db76588
#> Building 3 file(s):
#> Building analysis/about.Rmd
#> log directory created: /tmp/RtmpegncBz/workflowr
#> Building analysis/index.Rmd
#> Building analysis/license.Rmd
#> Summary from wflow_publish
#> 
#> **Step 1: Commit analysis files**
#> 
#> No files to commit
#> 
#> 
#> **Step 2: Build HTML files**
#> 
#> Summary from wflow_build
#> 
#> Settings:
#>  clean_fig_files: TRUE
#> 
#> The following were built externally each in their own fresh R session: 
#> 
#> docs/about.html
#> docs/index.html
#> docs/license.html
#> 
#> Log files saved in /tmp/RtmpegncBz/workflowr
#> 
#> **Step 3: Commit HTML files**
#> 
#> Summary from wflow_git_commit
#> 
#> The following was run: 
#> 
#>   $ git add docs/about.html docs/index.html docs/license.html docs/figure/about.Rmd docs/figure/index.Rmd docs/figure/license.Rmd docs/site_libs docs/.nojekyll 
#>   $ git commit -m "Build site." 
#> 
#> The following file(s) were included in commit 3411c8c:
#> docs/about.html
#> docs/index.html
#> docs/license.html
#> docs/site_libs/bootstrap-3.3.5/
#> docs/site_libs/highlightjs-9.12.0/
#> docs/site_libs/jquery-1.11.3/
#> docs/site_libs/navigation-1.1/
git2r::rm_file(path = "analysis/about.Rmd")
git2r::commit(message = "delete file")
#> [c30614d] 2020-03-02: delete file
file.create("analysis/about.Rmd")
#> [1] TRUE

# The file can be built
wflow_build("analysis/about.Rmd")
#> Current working directory: /tmp/RtmpegncBz/file7b604db76588
#> Building 1 file(s):
#> Building analysis/about.Rmd
#> Summary from wflow_build
#> 
#> Settings:
#> 
#> 
#> The following were built externally each in their own fresh R session: 
#> 
#> docs/about.html
#> 
#> Log files saved in /tmp/RtmpegncBz/workflowr
# Running wflow_status() provides a warning message and how to solve the issue
wflow_status("analysis/about.Rmd")
#> Warning: The following R Markdown file(s) have not been committed to the Git
#> repository but their corresponding HTML file(s) have. This violates the
#> reproducibility guarantee of workflowr. Please publish these files
#> using wflow_publish() to fix this situation.
#> 
#> analysis/about.Rmd
#> Status of 1 Rmd files
#> 
#> Totals:
#>  1 Scratch
#> 
#> The following Rmd files require attention:
#> 
#> Scr analysis/about.Rmd
#> 
#> Key: Scr = Scratch (Untracked)
#> 
#> The current Git status is:
#> 
#>     status substatus               file
#>  untracked untracked analysis/about.Rmd
#> 
#> To publish your changes as part of your website, use `wflow_publish()`.
#> To commit your changes without publishing them yet, use
#> `wflow_git_commit()`.
# Running wflow_publish() resolves the issue
wflow_publish("analysis/about.Rmd")
#> Warning: The following R Markdown file(s) have not been committed to the Git
#> repository but their corresponding HTML file(s) have. This violates the
#> reproducibility guarantee of workflowr. Please publish these files
#> using wflow_publish() to fix this situation.
#> 
#> analysis/about.Rmd
#> Current working directory: /tmp/RtmpegncBz/file7b604db76588
#> Building 1 file(s):
#> Building analysis/about.Rmd
#> Summary from wflow_publish
#> 
#> **Step 1: Commit analysis files**
#> 
#> Summary from wflow_git_commit
#> 
#> The following was run: 
#> 
#>   $ git add analysis/about.Rmd 
#>   $ git commit -m "wflow_publish(\"analysis/about.Rmd\")" 
#> 
#> The following file(s) were included in commit 47790e3:
#> analysis/about.Rmd
#> 
#> **Step 2: Build HTML files**
#> 
#> Summary from wflow_build
#> 
#> Settings:
#>  clean_fig_files: TRUE
#> 
#> The following were built externally each in their own fresh R session: 
#> 
#> docs/about.html
#> 
#> Log files saved in /tmp/RtmpegncBz/workflowr
#> 
#> **Step 3: Commit HTML files**
#> 
#> Summary from wflow_git_commit
#> 
#> The following was run: 
#> 
#>   $ git add docs/about.html docs/figure/about.Rmd docs/site_libs docs/.nojekyll 
#>   $ git commit -m "Build site." 
#> 
#> The following file(s) were included in commit 09f147e:
#> docs/about.html
#> docs/site_libs/jqueryui-1.11.4/
#> docs/site_libs/tocify-1.9.1/
wflow_status("analysis/about.Rmd")
#> Status of 1 Rmd files
#> 
#> Totals:
#>  1 Published
#> 
#> The current Git status is: working directory clean
#> 
#> Rmd files are up-to-date

# Cleanup
setwd(cwd)
unlink(d, recursive = TRUE)
```

<sup>Created on 2020-03-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

<details>

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.3 (2020-02-29)
#>  os       Ubuntu 18.04.4 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/New_York            
#>  date     2020-03-02                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version     date       lib source                   
#>  assertthat    0.2.1       2019-03-21 [3] CRAN (R 3.5.3)           
#>  backports     1.1.5       2019-10-02 [3] CRAN (R 3.6.1)           
#>  callr         3.4.0.9000  2020-01-06 [1] local                    
#>  cli           2.0.2       2020-02-28 [3] CRAN (R 3.6.3)           
#>  crayon        1.3.4       2017-09-16 [3] CRAN (R 3.5.0)           
#>  digest        0.6.25      2020-02-23 [3] CRAN (R 3.6.3)           
#>  evaluate      0.14        2019-05-28 [3] CRAN (R 3.6.0)           
#>  fansi         0.4.1       2020-01-08 [3] CRAN (R 3.6.2)           
#>  fs            1.3.1.9000  2020-01-08 [1] Github (r-lib/fs@2fb543e)
#>  git2r         0.26.1.9000 2019-11-27 [1] local                    
#>  glue          1.3.1       2019-03-12 [3] CRAN (R 3.5.3)           
#>  highr         0.8         2019-03-20 [3] CRAN (R 3.5.3)           
#>  htmltools     0.4.0       2019-10-04 [3] CRAN (R 3.6.1)           
#>  httpuv        1.5.2       2019-09-11 [3] CRAN (R 3.6.1)           
#>  knitr         1.28        2020-02-06 [3] CRAN (R 3.6.2)           
#>  later         1.0.0       2019-10-04 [3] CRAN (R 3.6.1)           
#>  magrittr      1.5         2014-11-22 [3] CRAN (R 3.5.0)           
#>  processx      3.4.1       2019-07-18 [3] CRAN (R 3.6.1)           
#>  promises      1.1.0       2019-10-04 [3] CRAN (R 3.6.1)           
#>  ps            1.3.2       2020-02-13 [3] CRAN (R 3.6.2)           
#>  R6            2.4.1       2019-11-12 [3] CRAN (R 3.6.1)           
#>  Rcpp          1.0.3       2019-11-08 [3] CRAN (R 3.6.1)           
#>  reprex        0.3.0.9000  2019-07-19 [1] local                    
#>  rlang         0.4.4       2020-01-28 [3] CRAN (R 3.6.2)           
#>  rmarkdown     2.1         2020-01-20 [1] CRAN (R 3.6.2)           
#>  rprojroot     1.2         2017-01-16 [3] CRAN (R 3.5.0)           
#>  rstudioapi    0.11        2020-02-07 [3] CRAN (R 3.6.2)           
#>  sessioninfo   1.1.1       2018-11-05 [3] CRAN (R 3.5.1)           
#>  stringi       1.4.6       2020-02-17 [3] CRAN (R 3.6.2)           
#>  stringr       1.4.0       2019-02-10 [3] CRAN (R 3.5.2)           
#>  withr         2.1.2       2018-03-15 [3] CRAN (R 3.5.0)           
#>  workflowr   * 1.6.0.9000  2020-03-02 [1] local                    
#>  xfun          0.11        2019-11-12 [3] CRAN (R 3.6.1)           
#>  yaml          2.2.1       2020-02-01 [3] CRAN (R 3.6.2)           
#> 
#> [1] /home/jdb-work/R/x86_64-pc-linux-gnu-library/3.6
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>
